lib/redis_memo/memoize_query/cached_select.rb

```diff
@@ -336,17 +337,18 @@ def self.extract_bind_params_recurse(node)
       end
 
       bind_params
-
-    when Arel::Nodes::NotEqual
-      # We don't cache based on NOT queries (where.not) because it is unbound
-      # but we memoize queries with NOT and other bound queries, so we return the original bind_params
-      return bind_params
     else
       # Not yet supported
       return
     end
   end
 
+  # Whether the expression of the node has the compare operators
+  def self.is_comparison_expr(expr)
+    return false if !expr.is_a?(Arel::Nodes::SqlLiteral)
+    %w[> >= < <= !=].any?{ |comparator| expr.to_s.include?(comparator) }
```

Comment:
Do you mean
```suggestion
    %w[> >= < <= !=].include?(expr.to_s)
```

Reply:
this has been removed, but it was the case that `expr` is `a > 3` so we wanted to see if the `expr` contains comparators in the list